### PR TITLE
Support stepped range syntax

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: test/ex_doc/autolink_test.exs [hunk 1]
```diff
@@ -106,6 +108,12 @@ defmodule ExDoc.AutolinkTest do
 
       assert autolink("for/1", apps: [:elixir]) ==
                ~m"[`for/1`](Kernel.SpecialForms.html#for/1)"
+
+      # TODO: Remove check once Elixir v1.12+ is required
+      if Version.compare(@elixir_version, "1.12.0-rc.0") in [:gt, :eq] do
```
Comment (josevalim) [inline test/ex_doc/autolink_test.exs:113]:
```suggestion
      if Version.match?(System.version(), ">= 1.12.0-rc.0") do
```

File: test/ex_doc/autolink_test.exs [hunk 2]
```diff
@@ -3,6 +3,8 @@ defmodule ExDoc.AutolinkTest do
   doctest ExDoc.Autolink
   import ExUnit.CaptureIO
 
+  @elixir_version System.build_info()[:version]
```
Comment (josevalim) [inline test/ex_doc/autolink_test.exs:6]:
```suggestion
```

File: test/ex_doc/autolink_test.exs [hunk 3]
```diff
@@ -3,6 +3,8 @@ defmodule ExDoc.AutolinkTest do
...
+
```
Comment (josevalim) [inline test/ex_doc/autolink_test.exs:7]:
```suggestion
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by eksperimental** [comment]
Much better @josevalim 
Thank you!

**Comment by eksperimental** [comment]
@josevalim we should release a new minor version of ExDoc before Elixir v1.12.0-rc.0 is out, so it will make use of this change.

**Pull request merged**